src/FsAutoComplete/LspServers/AdaptiveFSharpLspServer.fs

```diff
@@ -2602,26 +2607,22 @@ type AdaptiveFSharpLspServer
                     let items =
                       decls
                       |> Array.mapi (fun id d ->
-                        let code =
-                          if
-                            System.Text.RegularExpressions.Regex.IsMatch(d.NameInList, """^[a-zA-Z][a-zA-Z0-9']+$""")
-                          then
-                            d.NameInList
-                          elif d.NamespaceToOpen.IsSome then
-                            d.NameInList
-                          else
-                            FSharpKeywords.NormalizeIdentifierBackticks d.NameInList
-
-                        let label =
+                        let code = getCodeToInsert d
+
+                        let label, filterText =
                           match d.NamespaceToOpen with
-                          | Some no -> sprintf "%s (open %s)" d.NameInList no
-                          | None -> d.NameInList
+                          | Some no when config.FullNameExternalAutocomplete ->
+                            // allow filter "Ceiling (System.Math)" by "System.Math.Ceiling" or "CeilingSystem.Math"
+                            sprintf "%s (%s)" d.NameInList no, d.NameInList + code
+                          | Some no -> sprintf "%s (open %s)" d.NameInList no, d.NameInList
+                          | None -> d.NameInList, d.NameInList
```

Comment:
can you describe in a comment the logic going on here? how does the label selection and the filter text impact the user experience of searching/filtering in the IDE?

The previous implementation didn't document _why_ decisions were made, and so we don't have any sort of record for justification for making changes.